tip.lic

```diff
@@ -0,0 +1,99 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#tip
+=end
+
+# Background: I've always been annoyed by the need to specify tips in coppers.
+# DR's required syntax leaves you counting zeroes if you want to tip someone.
+# So a tip of 10 platinum kronars requires you to tip 100000 kronars. How many
+# zeroes is that? I lost count already. This simple script allows you to tip 
+# someone, in a similar syntax, but use the denomination to do so (e.g. "platinum").
+# <;tip gildaren 10 plat kron> now works. If you don't specify a currency,
+# this script also chooses the currency of the greatest coin you have on hand.
+
+custom_require.call %w[common common-money]
+
+class Tip
+
+  # include DRC for the message function
+  # include DRCM for use of converting amounts to copper
+  
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'recipient', regex: /^[A-z\.\s\-']+$/i, variable: true, description: 'The recipent of your tip. Abbreviations ok.' },
+        { name: 'amount', regex: /^\d+$/i, variable: true, description: 'Numeric amount to tip, no commas (e.g. the number "100" in 100 kronars)' },
+        { name: 'denomination', regex: /(\bp(l|la|lat|lati|latin|latinu|latinum)?\b)|(\bg(o|ol|old)?\b)|(\bs(i|il|ilv|ilve|ilver)?\b)|(\bb(r|ro|ron|ronz|ronze)?\b)|(\bc(o|op|opp|oppe|opper)?\b)/i, variable: true, description: 'The denomination. Abbrevations ok. (e.g. plat, gold, silv)' },
```

Comment:
I swear we've done something like this before... I just can't remember where.

@KatoakDR @Dartellum @jayRyan24 anyone else remember this?

Reply:
Safe-room has a tipping method I believe. A setting in base exists.

Reply:
Just checked it out. If I'm reading it right, it's a tip method for a PC empath(s) of known name(s), embedded within the safe-room script. It also uses the give verb instead of the tip verb.